app/cdap/components/hydrator/components/TopPanel/NameAndDescription/index.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  CancelButton,
+  CustomTooltip,
+  DescriptionTextField,
+  HydratorMetadata,
+  MetadataLeft,
+  NameTextField,
+  PipelineDescription,
+  PipelineName,
+  PipelineType,
+  SaveButton,
+  StyledIconSVG,
+} from './styles';
+
+interface IGlobalObj {
+  etlRealtime?: string;
+  etlDataStreams?: string;
+  etlBatchPipelines?: string[];
+}
+
+interface IMetadata {
+  description: string;
+  name: string;
+}
+
+interface IArtifact {
+  name: string;
+  scope: string;
+  version: string;
+}
+
+interface INameState {
+  metadata?: IMetadata;
+  artifact?: IArtifact;
+  viewSettings?: boolean;
+}
+
+interface INameAndDescriptionProps {
+  metadataExpanded?: boolean;
+  globals?: IGlobalObj;
+  state?: INameState;
+  invalidName?: string;
+  parsedDescription?: string;
+  saveMetadata?: (event: any) => void;
```

Comment:
@njbriggs any preference on the type we should be using for these events? SyntheticEvent? MouseEventHandler ?

Reply:
I'd prefer we not pass events out of components - components should emit semantically meaningful data. That said, is this necessary to work with the Angular code for now?

Reply:
Yeah for now it's necessary, the Angular controller needs to handle the events